cognite/client/cognite_client.py

```diff
@@ -83,13 +84,15 @@ def __init__(
         timeout: int = None,
         debug: bool = None,
     ):
+        thread_local_api_key, thread_local_project = self._get_thread_local_credentials()
+
         environment_api_key = os.getenv("COGNITE_API_KEY")
```

Comment:
Should there be an `environment_project` as well?

Reply:
No, I don't think this is necessary since we autofetch the project in case it is not specified. I think it makes it easier to make mistakes which can result in 401s if we have that.